CMakeLists.txt

```diff
@@ -311,8 +314,17 @@ if(JANSSON_BUILD_SHARED_LIBS)
       ${JANSSON_HDR_PUBLIC}
       src/jansson.def)
 
-   if (LIBC_IS_GLIBC)
+   if (LIBC_IS_GLIBC AND LD_SYMVER)
       set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--default-symver")
+   else()
+      file(WRITE "${CMAKE_SOURCE_DIR}/jansson.map" "libjansson.so.${JANSSON_SOVERSION} {
```

Comment:
Any chance this can go in the build dir instead, to avoid writing files to the source dir? I'm not experienced with cmake, but it looks like that might be `${CMAKE_CURRENT_BINARY_DIR}`.

Also, this could probably just switch to `--version-script` unconditionally and it would work for all linkers.

Reply:
Sure, will update that.

Reply:
Ok done